tests/integration/test_integration_all_rust.py

```diff
@@ -590,983 +583,1003 @@ def teardown_module():
+    # to have two LogCheck entires and test_no_sentry_output would have data
+    # when it should not.
+
+    # new autoconnect emits 2 events
+    try:
+        MOCK_SENTRY_QUEUE.get(timeout=5)
+    except Empty:
+        pass
+    assert event1["exception"]["values"][0]["value"] == "LogCheck"
+
+    process_logs_autouse(max_conn_logs=4)
+
+
+async def test_sentry_output_autoendpoint(
+    registered_test_client: AsyncPushTestClient, process_logs_autouse
+) -> None:
+    """Test sentry output for autoendpoint."""
+    if os.getenv("SKIP_SENTRY"):
+        log.debug("Skipping test_sentry_output_autoendpoint")
+        pytest.skip("Skipping test_sentry_output_autoendpoint")
+    endpoint = registered_test_client.get_host_client_endpoint()
+    await registered_test_client.disconnect()
+    async with httpx.AsyncClient() as httpx_client:
+        await httpx_client.get(f"{endpoint}/__error__", timeout=5)
+        # 2 events excpted: 1 from a panic and 1 from a returned Error
+
+    event1 = MOCK_SENTRY_QUEUE.get(timeout=5)
+    event2 = MOCK_SENTRY_QUEUE.get(timeout=5)
+    values = (
+        event1["exception"]["values"][0]["value"],
+        event2["exception"]["values"][0]["value"],
+    )
+    # "ERROR:Success" & "LogCheck" are the commonly expected values.
+    # When updating to async/await, noted cases in CI where only LogCheck
+    # occured, likely due to a race condition.
+    # Establishing a single check for "LogCheck" as sufficent guarantee,
+    # since Sentry is capturing emission
+    assert "LogCheck" in values
+    assert sorted(values) == ["ERROR:Success", "LogCheck"]
+    process_logs_autouse(max_endpoint_logs=1)
+
+
+async def test_no_sentry_output(ws_url: str, process_logs_autouse) -> None:
+    """Test for no Sentry output."""
+    if os.getenv("SKIP_SENTRY"):
+        log.debug("Skipping test_no_sentry_output")
+        pytest.skip("Skipping test_no_sentry_output")
+
+    ws_url = urlparse(ws_url)._replace(scheme="http").geturl()
+
+    try:
         async with httpx.AsyncClient() as httpx_client:
-            await httpx_client.get(f"{endpoint}/__error__", timeout=5)
-            # 2 events excpted: 1 from a panic and 1 from a returned Error
-
-        event1 = MOCK_SENTRY_QUEUE.get(timeout=5)
-        event2 = MOCK_SENTRY_QUEUE.get(timeout=5)
-        values = (
-            event1["exception"]["values"][0]["value"],
-            event2["exception"]["values"][0]["value"],
+            await httpx_client.get(ws_url, timeout=30)
+            data = MOCK_SENTRY_QUEUE.get(timeout=5)
+            assert not data
+    except httpx.ConnectError:
+        pass
+    except Empty:
+        pass
+    process_logs_autouse(max_conn_logs=4)
+
+
+async def test_hello_echo(test_client) -> None:
+    """Test hello echo."""
+    await test_client.connect()
+    result = await test_client.hello()
+    assert result != {}
+    assert result["use_webpush"] is True
+    await test_client.disconnect()
```

Comment:
The `test_client` fixture calls `disconnect`, do we also need to do it here?

Question also applies to: 
- test_hello_with_bad_prior_uaid

Reply:
Good catch, there are some explicit `disconnect` calls, but these are not actually required in this case since it will auto disconnect.